core/src/templates/components/media/media.twig

```diff
@@ -0,0 +1,51 @@
+{#
+/**
+ * @file
+ * Media Component
+ *
+ * A media element, e.g., image, video, audio, iframe etc.
+ *
+ * Available variants:
+ * .su-media--image - Image
+ * .su-media--video - HTML5 video
+ * .su-media--audio - HTML5 audio
+ *
+ * Available variables:
+ * - attributes: For additional HTML attributes not already provided.
+ * - modifier_class: Additional CSS classes to change look and behavior.
+ * - media_wrapper_modifier_class: Additional CSS classes to change look and behavior of the media wrapper, e.g. "su-aspect-ratio" to provide aspect ratio constraint.
+ * - media_image_src: The src link to the image file
+ * - media_image_srcset: The srcset of the image if exists
+ * - media_video_src: The src link to the video file
+ * - media_video_poster: The preview image of the video
+ * - media_audio_src: The src link to the audio file
+ * - media_custom: An open field for the full HTML element of the media content, e.g., an embedded YouTube iframe
+ * - media_content_attributes: Additional HTML attributes for the media content, e.g., for <video> - controls, autoplay; for <img> - alt, width, height
+ * - media_caption: Caption for the media.
+ */
+#}
+<figure {{ attributes }} class="su-media {{ modifier_class }}">
+    <div class="su-media__wrapper {{ media_wrapper_modifier_class }}">
+      {% if 'su-media--video' in modifier_class %}
+        <video src="{{ media_video_src }}"
+          {% if media_video_poster %}
+            poster="{{ media_video_poster }}"
+          {% endif %}
+        {{ media_content_attributes }}></video>
+      {% elseif 'su-media--audio' in modifier_class %}
+        <audio src="{{ media_audio_src }}" {{ media_content_attributes }} controls></audio>
+      {% elseif 'su-media--image' in modifier_class %}
+        <img src="{{ media_image_src }}"
+            {% if media_image_srcset %}
+              srcset="{{ media_image_srcset }}"
+            {% endif %}
+        {{ media_content_attributes }}>
+      {% else %}
```

Comment:
The markup for this is different for us and the srcset looks like the following:

```
<picture>
  <source srcset="/image/photo-epel-stanford-campus_1.jpg?itok=lwFxUYbB 1x, /image/photo-epel-stanford-campus_1.jpg?itok=lwFxUYbB 3000w, /image/photo-epel-stanford-campus_1.jpg 4033w" media="screen and (min-width: 1200px)" type="image/jpeg" sizes="100vw">
<source srcset="/image/photo-epel-stanford-campus_1.jpg?itok=aRtBp3Zw 1x, /image/photo-epel-stanford-campus_1.jpg?itok=MdHIwjgR 2048w, /image/photo-epel-stanford-campus_1.jpg?itok=lwFxUYbB 3000w" media="screen and (min-width: 992px)" type="image/jpeg" sizes="100vw">
<source srcset="/image/photo-epel-stanford-campus_1.jpg?itok=oAqwDanx 1x, /image/photo-epel-stanford-campus_1.jpg?itok=fbf5BsRc 1536w, /image/photo-epel-stanford-campus_1.jpg?itok=MdHIwjgR 2048w" media="screen and (min-width: 768px)" type="image/jpeg" sizes="100vw">
<source srcset="/image/photo-epel-stanford-campus_1.jpg?itok=Iqlg1rLJ 1x, /image/photo-epel-stanford-campus_1.jpg?itok=2JxhTWX7 1152w, /image/photo-epel-stanford-campus_1.jpg?itok=fbf5BsRc 1536w" media="screen and (min-width: 576px)" type="image/jpeg" sizes="100vw">
<source srcset="/image/photo-epel-stanford-campus_1.jpg?itok=gkwi4Znj 1x, /media/image/photo-epel-stanford-campus_1.jpg?itok=2JxhTWX7 1152w, /image/photo-epel-stanford-campus_1.jpg?itok=fbf5BsRc 1536w" media="(min-width: 0em)" type="image/jpeg" sizes="100vw">
<img src="/image/photo-epel-stanford-campus_1.jpg?itok=lwFxUYbB" role="presentation">
</picture>
```

Reply:
Yeah, I did notice that when testing the D8. I'll test this out to make sure it works with this component.

Reply:
Perhaps pull image out as that is a whole different beast on its own?

Reply:
Would your srcset code work with the default which takes open HTML {{media_custom}}?

Reply:
@sherakama , I'm going to do some thinking and see what's the best way to do this so the <picture> element would work with the aspect ratio mixin too. If it gets too complicated I'm fine with it being a different component 😄

Reply:
@sherakama , how would you prefer the variable for the srcset to be in order to work best with D8? For example, all the `<source>` tags in one big variable, or as an array, or an array of all the `srcset` + an array of other attributes (media, sizes, type) for each source tag? `<video>` and `<audio>` could take different sources too - should I make that an option also? To make it a little less complicated, I could also make a .su-media--picture variant.

Reply:
D8 is not going to use this component. If someone needs to pass in srcset in the above format, they can use the `{{ media_custom }}` variable which takes open HTML.